docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -0,0 +1,222 @@
+---
+title: Integrate Node OS Image caching with CAPZ
+authors:
+  - "@bennycortese"
+reviewers:
+  - "@nojnhuh"
+  - "@CecileRobertMichon"
+  - "@jackfrancis"
+  - "@willie-yao"
+creation-date: 2023-06-12
+last-updated: 2023-06-12
+
+status: provisional
+see-also:
+  - "/docs/proposals/20190101-we-heard-you-like-proposals.md"
+  - "/docs/proposals/20190102-everyone-gets-a-proposal.md"
+replaces:
+  - "/docs/proposals/20181231-replaced-proposal.md"
+superseded-by:
+  - "/docs/proposals/20190104-superceding-proposal.md"
+---
+
+# Title 
+Integrate Node OS Image caching with CAPZ
+
+## Table of Contents
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Requirements (Optional)](#requirements-optional)
+      - [Functional Requirements](#functional-requirements)
+        - [FR1](#fr1)
+        - [FR2](#fr2)
+      - [Non-Functional Requirements](#non-functional-requirements)
+        - [NFR1](#nfr1)
+        - [NFR2](#nfr2)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Node Prototype Pattern - Where we take a known good/working OS snapshot and make a “prototype” upon which all future node OS’ images are based.
+
+Warm nodes - The concept of keeping up extraneous and unused nodes to prevent having to wait for patches or security updates, thus allowing you to have those nodes ready when more users try to use your service without the wait times.
+
+Shared Image Gallery - A repository that is an Azure Resource for managing and sharing images
+
+Snapshot - A static copy of the image at the point in time that it is taken
+
+## Summary
+
+A controller will be added to CAPZ which caches the Nodes’ OS image on a regular interval and updates the model to use that image for future scale outs. This feature request will allow for faster horizontal scaling and will help prevent security alerts from new nodes which immediately need a security update as those new nodes will now come with the security update already installed. This feature will also help prevent users from needing to spin up warm nodes due to this faster horizontal scaling.
```

Comment:
It could be beneficial to break the sentences about what the feature will allow for into bullet points.

Reply:
It was, fixed, thanks!